chia/util/paginator.py

```diff
@@ -0,0 +1,37 @@
+import dataclasses
+from math import ceil
+from typing import Sequence, Tuple
+
+
+class InvalidPageSizeError(Exception):
+    def __init__(self, page_size: int) -> None:
+        super().__init__(f"Invalid page size {page_size}. Available sizes: {Paginator.page_sizes}")
+
+
+class PageOutOfBoundsError(Exception):
+    def __init__(self, page_size: int, max_page_size: int) -> None:
+        super().__init__(f"Page {page_size} out of bounds. Available pages: 1-{max_page_size}")
+
+
+@dataclasses.dataclass
+class Paginator:
+    _source: Sequence[object]
+    _page_size: int
+
+    page_sizes: Tuple[int, ...] = (5, 10, 25, 50, 100)
```

Comment:
I don't think this restriction improves the class.

Reply:
See [5d1e2a1](https://github.com/Chia-Network/chia-blockchain/pull/11247/commits/5d1e2a1a6bd8b1f3b1adfb6252e45ef205c6597e)